### PR TITLE
[Python3 migration]Fix test failure in dual_tor_io.py

### LINKED ISSUE #8021: dualtor_io tests fail for converting to python3
**Description**

dualtor_io test failed after converting to python3: 
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <Raw  load='b\'\\x00\\xaa\\xbb\\xcc\\xdd\\xee\\x00\\x06\\x07\\x08\\t\\n\\x08\\x00E\\x00\\x00V\\x00\\x01\\x00\\x00@\\x0...x0c\\r\\x0e\\x0f\\x10\\x11\\x12\\x13\\x14\\x15\\x16\\x17\\x18\\x19\\x1a\\x1b\\x1c\\x1d\\x1e\\x1f !"#$%&\\\'()*+,-\'' |>
cls = <class 'scapy.layers.l2.Ether'>

    def __getitem__(self, cls):
        # type: (Union[Type[Packet], str]) -> Any
        if isinstance(cls, slice):
            lname = cls.start
            if cls.stop:
                ret = self.getlayer(cls.start, nb=cls.stop, **(cls.step or {}))
            else:
                ret = self.getlayer(cls.start, **(cls.step or {}))
        else:
            lname = cls
            ret = self.getlayer(cls)
        if ret is None:
            if isinstance(lname, type):
                name = lname.__name__
            elif not isinstance(lname, bytes):
                name = repr(lname)
            else:
                name = cast(str, lname)
>           raise IndexError("Layer [%s] not found" % name)
E           IndexError: Layer [Ether] not found

cls        = <class 'scapy.layers.l2.Ether'>
lname      = <class 'scapy.layers.l2.Ether'>
name       = 'Ether'
ret        = None
self       = <Raw  load='b\'\\x00\\xaa\\xbb\\xcc\\xdd\\xee\\x00\\x06\\x07\\x08\\t\\n\\x08\\x00E\\x00\\x00V\\x00\\x01\\x00\\x00@\\x0...x0c\\r\\x0e\\x0f\\x10\\x11\\x12\\x13\\x14\\x15\\x16\\x17\\x18\\x19\\x1a\\x1b\\x1c\\x1d\\x1e\\x1f !"#$%&\\\'()*+,-\'' |>

/var/AzDevOps/env-python3/lib/python3.8/site-packages/scapy/packet.py:1344: IndexError
```

**Steps to reproduce the issue:**
1.
2.
3.

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important:**

    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

### PR DESCRIPTION
**Pull request by wsycqyz**
### Description of PR

Summary:
Fixes #8021 dualtor_io tests fail for converting to python3

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
dualtor_io test failed after converting to python3 migration with error:
E           IndexError: Layer [Ether] not found

#### How did you do it?
In Python2, str() is used to process packet.
In Python3, bytes() should be used.

#### How did you verify/test it?
Manually run test case.

### CODE REVIEW

File: tests/common/dualtor/dual_tor_io.py [hunk 1]
```diff
@@ -293,7 +294,10 @@ def generate_downstream_traffic(self, dst='server'):
             ip_ttl=ip_ttl,
             tcp_dport=TCP_DST_PORT
         )
-        tcp_tx_packet_orig = scapyall.Ether(str(tcp_tx_packet_orig))
+        if six.PY2:
```
Comment (lolyu) [inline tests/common/dualtor/dual_tor_io.py:297]: Can we wrap this logic into a helper function:
```suggestion
def convert_scapy_packet_to_bytes(packet):
    if six.PY2:
        return str(packet)
    else:
        return bytes(packet)
```
And maybe it is better to store it in a common module like `tests/common/scapy_utilities.py`

**Review by lolyu — COMMENTED** [review]
Thanks @wsycqyz for the fix!

File: tests/common/dualtor/dual_tor_io.py (hunk 1 above)
Comment (wsycqyz) [reply tests/common/dualtor/dual_tor_io.py:297]: Will wrap this logic into a helper function.
Will commit later after I finish running all test cases under dualtor_io.

File: tests/common/dualtor/dual_tor_io.py [hunk 2]
```diff
@@ -586,13 +586,19 @@ def examine_flow(self):
             server_addr = self.get_server_address(packet)
             server_to_packet_map[server_addr].append(packet)
 
+        # E731 Use a def instead of a lambda
+        def get_packet_sort_key(packet):
+            if six.PY2:
+                payload_bytes = str(packet[scapyall.TCP].payload)
```
Comment (lolyu) [inline tests/common/dualtor/dual_tor_io.py:592]:
```suggestion
                payload_bytes = convert_scapy_packet_to_bytes(packet[scapyall.TCP].payload)
```

File: tests/common/dualtor/dual_tor_io.py [hunk 3]
```diff
@@ -627,7 +633,10 @@ def examine_each_packet(self, server_ip, packets):
                 # scapy 2.4.5 will use Decimal to calulcate time, but json.dumps
                 # can't recognize Decimal, transform to float here
                 curr_time = float(packet.time)
-                curr_payload = int(str(packet[scapyall.TCP].payload).replace('X', ''))
+                if six.PY2:
```
Comment (lolyu) [inline tests/common/dualtor/dual_tor_io.py:636]: Same as above

File: tests/common/dualtor/dual_tor_io.py [hunk 4]
```diff
@@ -704,8 +713,12 @@ def check_tcp_payload(self, packet):
             sequential TCP Payload
         """
         try:
-            int(str(packet[scapyall.TCP].payload).replace('X', '')) in range(
-                self.packets_to_send)
+            if six.PY2:
```
Comment (lolyu) [inline tests/common/dualtor/dual_tor_io.py:716]: Same as above

File: tests/common/dualtor/dual_tor_io.py (hunk 2 above)
Comment (wsycqyz) [reply tests/common/dualtor/dual_tor_io.py]: Accepted. Thanks!

File: tests/common/dualtor/dual_tor_io.py (hunk 3 above)
Comment (wsycqyz) [reply tests/common/dualtor/dual_tor_io.py:637]: Accepted. Thanks!

File: tests/common/dualtor/dual_tor_io.py (hunk 4 above)
Comment (wsycqyz) [reply tests/common/dualtor/dual_tor_io.py:718]: Accepted. Thanks!

**Review by lolyu — APPROVED** [review]

**Pull request merged**